BUILD.md

```diff
@@ -14,6 +14,16 @@ up builds. The minimum set of dependencies is:
 All other dependencies are optional and can be disabled while configuring the
 build (in `meson setup` step).
 
+## General notes
+
+Meson allows maintaining different build configurations for a number of build
+targets in their own separate subdirectories. E.g. you can keep you debug and
+release configurations in the `build/debug` and `build/release` directories,
+respectively.
+
+If you want to reconfigure an existing target with `meson setup <build_path>`,
+make sure to delete the directory `<build_path>` first.
```

Comment:
There's no need to do that. First, Meson tries its best to clean up build directory as necessary, so most of the time e.g. switching from debug to release configuration, flipping flags, and even jumping between tags and branches won't cause any problems (in fact, that's the way I usually operate). Another option is to reconfigure, then call `meson setup <build_path> --wipe` to set the build directory up from scratch, but keeping the configuration.

Reply:
Ah, thanks. Somehow I was under the impression that you had to do this. Removing this whole sentence altogether then.